runtime/compiler/z/codegen/J9TreeEvaluator.hpp

```diff
@@ -115,6 +115,31 @@ class OMR_EXTENSIBLE TreeEvaluator: public J9::TreeEvaluator
    static TR::Register *conditionalHelperEvaluator(TR::Node *node, TR::CodeGenerator *cg);
    static float interpreterProfilingInstanceOfOrCheckCastTopProb(TR::CodeGenerator * cg, TR::Node * node);
 
+   // Software Concurrent Scavenge evaluators
+   static TR::Register *ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg);
+   /**
+    * \brief
+    * In concurrent scavenge (CS) mode, this is used to evaluate ardbari nodes and inlined-compare-and-swap to
+    * generate a guarded load software sequence that is functionally equivalent to LGG/LLGFSG instructions on z.
+    *
+    * \param node
+    * \param cg
+    * \param resultReg
+    * \param memRef
+    * \param isCompressedRef
+    * \param hasReturnReg
+    * \param dynLitPooLoad
+    *
+    * \return NULL if the hasReturnReg parameter is true; otherwise, it returns a register that contains the reference field.
+   */
+   static TR::Register *generateSoftwareReadBarrier(TR::Node* node,
+                                                    TR::CodeGenerator* cg,
+                                                    TR::Register* resultReg,
+                                                    TR::MemoryReference* memRef,
+                                                    const bool shouldDecompressField,
```

Comment:
Do we need this parameter? Why not just use `cg->comp()->useCompressedPointers()`?

Reply:
Fixed.